src/components/presentational/Chat/Chat.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+import { faPaperPlane } from '@fortawesome/free-solid-svg-icons/faPaperPlane';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons/faSpinner';
+import MarkdownIt from 'markdown-it';
+import parse, { DOMNode, domToReact, Element } from 'html-react-parser';
+
+import UnstyledButton from '../UnstyledButton';
+
+import './Chat.css';
+
+export type ChatMessageType = "sent" | "received";
+
+export interface ChatProps {
+    messages: ChatMessage[];
+    onSendMessage: (newMessage: string) => void;
+    loading?: string;
+}
+
+export interface ChatMessage {
+    icon: React.JSX.Element;
+    content: string;
+    type: ChatMessageType;
+}
+
+export default function (props: ChatProps) {
+
+    let md = new MarkdownIt();
+
+    const [currentUserMessage, setCurrentUserMessage] = useState('');
+    const logRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        if (logRef.current) {
+            logRef.current.scrollTop = logRef.current.scrollHeight;
+        }
+    }, [props.messages]);
+
+    const sendMessage = async function () {
+        let newMessage = currentUserMessage;
+        props.onSendMessage(newMessage);
+        setCurrentUserMessage('');
+    }
+
+    return (
+        <div className="mdhui-chat">
+            <div className="mdhui-chat-log" ref={logRef}>
+                {props.messages.map((message, index) => (
+                    <div key={index} className={"mdhui-chat-message " + message.type}>
+                        {message.icon}{parse(md.render(message.content), {
+                            transform: (reactNode, domNode, index) => {
+                                if (domNode.nodeType === 1 && domNode.name === "p" && index === 0) {
```

Comment:
What's the point of this?

Reply:
Crazy ninja code to inline the icon. Will probably end up removing it.